src/managers/SXCursorManager.hpp

```diff
@@ -0,0 +1,205 @@
+#pragma once
+#include <string>
+#include <vector>
+#include <array>
+#include <cstdint>
+#include <hyprutils/math/Vector2D.hpp>
+#include "helpers/memory/Memory.hpp"
+
+extern "C" {
+#include <X11/Xcursor/Xcursor.h>
+}
+
+// gangsta bootleg XCursor impl. adidas balkanized
+struct SXCursorImage {
+    Vector2D              size;
+    Vector2D              hotspot;
+    std::vector<uint32_t> pixels; // XPixel is a u32
+    uint32_t              delay;  /* animation delay to next frame (ms) */
+};
+
+struct SXCursors {
+    std::vector<SXCursorImage> images;
+    std::string                shape;
+};
+
+// Taken from https://gitlab.freedesktop.org/xorg/lib/libxcursor/-/blob/master/src/library.c
+// clang-format off
+static std::array<const char*, 77> XCURSOR_STANDARD_NAMES = {
+    "X_cursor",
+    "arrow",
+    "based_arrow_down",
+    "based_arrow_up",
+    "boat",
+    "bogosity",
+    "bottom_left_corner",
+    "bottom_right_corner",
+    "bottom_side",
+    "bottom_tee",
+    "box_spiral",
+    "center_ptr",
+    "circle",
+    "clock",
+    "coffee_mug",
+    "cross",
+    "cross_reverse",
+    "crosshair",
+    "diamond_cross",
+    "dot",
+    "dotbox",
+    "double_arrow",
+    "draft_large",
+    "draft_small",
+    "draped_box",
+    "exchange",
+    "fleur",
+    "gobbler",
+    "gumby",
+    "hand1",
+    "hand2",
+    "heart",
+    "icon",
+    "iron_cross",
+    "left_ptr",
+    "left_side",
+    "left_tee",
+    "leftbutton",
+    "ll_angle",
+    "lr_angle",
+    "man",
+    "middlebutton",
+    "mouse",
+    "pencil",
+    "pirate",
+    "plus",
+    "question_arrow",
+    "right_ptr",
+    "right_side",
+    "right_tee",
+    "rightbutton",
+    "rtl_logo",
+    "sailboat",
+    "sb_down_arrow",
+    "sb_h_double_arrow",
+    "sb_left_arrow",
+    "sb_right_arrow",
+    "sb_up_arrow",
+    "sb_v_double_arrow",
+    "shuttle",
+    "sizing",
+    "spider",
+    "spraycan",
+    "star",
+    "target",
+    "tcross",
+    "top_left_arrow",
+    "top_left_corner",
+    "top_right_corner",
+    "top_side",
+    "top_tee",
+    "trek",
+    "ul_angle",
+    "umbrella",
+    "ur_angle",
+    "watch",
+    "xterm",
+};
+
+static std::vector<uint32_t> HYPR_XCURSOR_PIXELS = {
```

Comment:
why is this in the header

Reply:
moved it back to the .cpp file, had it mostly in the header to not having to scroll those extra 100 lines when implenting heh.